### PR TITLE
Domain Management: UI tweaks

### PR DESCRIPTION
**Pull request by ramynasr**
This PR introduces some tweaks to help declutter the `Domain Management` screen.

Related to Automattic/nomado-issues#885

## Proposed Changes

* `Manage All Domains` button is removed from the bottom of the page to the top right corner.
* Hide the domains dropdown list when it's disabled. This happens when the site plan does not support setting a custom primary domain or when there are no domains to select from.
* Only show checkboxes in the domains table when Bulk actions make sense (when there are two or more selectable domains).

## Why are these changes being made?

* Checkboxes and bulk actions are irrelevant when there is only domain the user can act on.
* `Manage All Domains` CTA at the bottom of the page was distracting and cluttered when paired with the other CTAs.
* The disabled dropdown and button in the "Primary Site Address" card were adding non actionable elements to the UI.

## Testing Instructions

We need to check the domain management page within a site context ( `/domains/manage/{site-slug}` ) with the following scenarios and make sure the changes mentioned above work as expected:

* A free site with no custom domains.
* A free site with a custom domain that is either registered or mapped.
* A paid site with no custom domains.
* A paid site with at least one custom domain that is either registered or mapped.

## Screenshots

| Before | After |
| --- | --- |
| image | image |
| image | image |
| image | image |
| image | image |

### CODE REVIEW

File: client/my-sites/domains/domain-management/components/primary-domain-selector/index.tsx [hunk 1]
```diff
@@ -1,7 +1,7 @@
 import { recordTracksEvent } from '@automattic/calypso-analytics';
 import { FEATURE_SET_PRIMARY_CUSTOM_DOMAIN } from '@automattic/calypso-products';
 import { Card } from '@automattic/components';
-import { useTranslate } from 'i18n-calypso';
+import { Substitution, useTranslate } from 'i18n-calypso';
```
Comment (gius80) [inline client/my-sites/domains/domain-management/components/primary-domain-selector/index.tsx:4]:
```suggestion
import { type Substitution, useTranslate } from 'i18n-calypso';
```
nit

**Review by gius80 — APPROVED** [review]
LGTM! I've tested all the different combination and everything looks good. I'm not 100% about removing "The current primary address set for this site is..." paragraph, even if the primary domain cannot be changed, but I don't have a strong opinion honestly.

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/16413433

Some locales (Hebrew, Japanese) have been temporarily machine-translated due to translator availability. All other translations are usually ready within a few days. Untranslated and machine-translated strings will be sent for translation next Monday and are expected to be completed by the following Friday.

Thank you @ramynasr for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/16413433).

**Pull request merged**